grails-app/services/au/org/emii/aatams/detection/DetectionExtractService.groovy

```diff
@@ -59,39 +57,48 @@ class DetectionExtractService extends AbstractStreamingExporterService {
     protected void writeCsvData(final filterParams, OutputStream out)
     {
         filterParams.sql = new Sql(dataSource)
+        filterParams.sql.fetchSize = getFetchSize()
+        filterParams.sql.autoCommit = false
         filterParams.projectPermissionCache = [:]
 
         super.writeCsvData(filterParams, out)
     }
 
-    protected def readData(filterParams)
+    protected def eachRow(filterParams, closure)
     {
-        return extractPage(filterParams)
+        def startTime = System.currentTimeMillis()
+        def query =  new QueryBuilder().constructQuery(filterParams)
+        def releaseIsProtectedCache = [:]
+        int count = 0
+        
+        filterParams.sql.eachRow(query.getSQL(), query.getBindValues()) { row ->
+            def sanitisedRows = applyEmbargo([row.toRowResult()], filterParams, releaseIsProtectedCache)
+            sanitisedRows.each { sanitisedRow ->
+                closure.call(sanitisedRow)
+            }
+            count++
+        }
```

Comment:
Yes, this can be cleaned up a bit.  I was just focussing on getting the cursor streaming to work initially without changing the embargo class around so I just made it fit here.  I'll look at adding a isRestricted and sanitise or similar methods to the DetectionVisibility class to more clearly express what the processing is here.   

I think we should leave the grouping (fetching in batches) to the infrastructure and just iterate through rows here.  Our own grouping complicates the code.